File: Prueba y error/ejercicios de semana santa java/three java/Library/SegundaLey.java
package Library;

//  Segunda Ley de Newton (F = m * a)
public class SegundaLey extends LeyNewton {

    public SegundaLey() {
        super("Segunda Ley de Newton - F = m * a");
    }

    @Override
    public String explicarLey() {
        return "La fuerza es el producto de la masa por la aceleración: F = m * a.";
    }

    @Override
    public String calcular(double... valores) {
        if (valores.length < 2) {
            return "Error: se necesitan la masa y la aceleración para calcular la fuerza.";
        }
        double masa = valores[0];
        double aceleracion = valores[1];
        double fuerza = masa * aceleracion;
        return "Fuerza calculada: " + fuerza + " N (F = " + masa + " kg * " + aceleracion + " m/s²)";
    }
}

File: Prueba y error/ejercicios de semana santa php/Thre php/Library/SegundaLey.php
<?php
namespace Library;

require_once 'LeyNewton.php';

class SegundaLey extends leyNewton {
    private float $masa;
    private float $aceleracion;
    private float $fuerza;

    public function __construct(float $masa, float $aceleracion) {
        $this->masa = $masa;
        $this->aceleracion = $aceleracion;
        $this->fuerza = $this->masa * $this->aceleracion;
    }

    public function describir(): string {
        return "Segunda Ley de Newton - F = m * a";
    }

    public function ejecutar(): void {
        echo "===== " . $this->describir() . " =====<br>";
        echo "La fuerza es el producto de la masa por la aceleración: F = m * a.<br>";
        echo "Fuerza calculada: {$this->fuerza} N (F = {$this->masa} kg * {$this->aceleracion} m/s²)<br>";
    }
}